firedrake/preconditioners/hypre_ams.py

```diff
@@ -44,14 +43,8 @@ def initialize(self, obj):
         if zero_beta:
             pc.setHYPRESetBetaPoissonMatrix(None)
 
-        # Build constants basis for the Nedelec space
-        cvecs = []
-        for i in range(mesh.cell_dimension()):
-            direction = [1.0 if i == j else 0.0 for j in range(mesh.cell_dimension())]
-            c = project(Constant(direction), V)
-            with c.vector().dat.vec_ro as cvec:
-                cvecs.append(cvec)
-        pc.setHYPRESetEdgeConstantVectors(*cvecs)
+        VectorP1 = VectorFunctionSpace(mesh, "Lagrange", 1)
+        pc.setCoordinates(interpolate(SpatialCoordinate(mesh), VectorP1).dat.data_ro)
```

Comment:
I would prefer to take a copy of the coordinates and drop the `Function` (so tack a `.copy()` on the end here).